### PR TITLE
Style fix for Array builtin

### PR DESCRIPTION
**Pull request by bzsolt**
- fix some style issues
- sort the functions according by the chapters of the standard

Related issue: #110

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by ruben-ayrapetyan** [comment]
@bzsolt, could you, please, split the style fix into two commits in the pull request?
- sorting functions
- other style fixes

**Comment by bzsolt** [comment]
@ruben-ayrapetyan, I've splitted into two commits as you requested.

**Comment by ruben-ayrapetyan** [comment]
@bzsolt, thank you!

**Comment by ruben-ayrapetyan** [comment]
@bzsolt, could you, please, also update message of first commit (currently, "Fix for the Array builtin"), so that it would become clear, that the commit only reorders functions and other definitions?

**Comment by ruben-ayrapetyan** [comment]
After updating commit message and `index ? index-- : zero_reached = true;`-like expressions, the changes would look good to me.

**Comment by egavrin** [comment]
@bzsolt @ruben-ayrapetyan signed `index` should solve this.

**Comment by bzsolt** [comment]
@egavrin @ruben-ayrapetyan The PR is updated.

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp [hunk 1]
```diff
@@ -2745,57 +2742,59 @@ ecma_builtin_array_prototype_object_reduce_right (ecma_value_t this_arg, /**< th
       ecma_number_t *num_p = ecma_alloc_number ();
       ecma_value_t accumulator = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
 
-      /* 6 */
-      uint32_t index = len - 1;
-      bool zero_reached = !len;
+      /* 6. */
+      int32_t index = (int32_t) len - 1;
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: `int32_t` is not enough for case `len` is more than 2147483647 (that is allowed by specification). `int64_t` would be correct for all possible array indices.

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp [hunk 2]
```diff
@@ -2745,57 +2742,59 @@ ecma_builtin_array_prototype_object_reduce_right (ecma_value_t this_arg, /**< th
...
           if ((k_present = (ecma_op_object_get_property (obj_p, index_str_p) != NULL)))
           {
             ECMA_TRY_CATCH (current_value, ecma_op_object_get (obj_p, index_str_p), ret_value);
             accumulator = ecma_copy_value (current_value, true);
             ECMA_FINALIZE (current_value);
           }
-          /* 8b-iv */
-          index ? --index : zero_reached = true;
+          /* 8.b.iv */
+          if (index != 0)
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: As `index` is a signed integer, the check is not necessary.

Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: Test case:

```
var func = function(a, b) {
  return a + b;
}


a = new Array();
a.length = 10;
a.reduceRight (func);
```

**Comment by ruben-ayrapetyan** [comment]
@bzsolt, thank you for update

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

**Comment by galpeter** [comment]
looks good to me

**Comment by egavrin** [comment]
Merged cdf361c2bd2fd6d0a014c665e82b820f0c96cc85 352e8f5c04b2d9595842b435360b1c927a3e3b63

**Pull request closed without merging**